src/shared/ui/composables/useApplicationCurrency.ts

```diff
@@ -0,0 +1,22 @@
+import { computed, ref } from 'vue';
+
+export const APPLICATION_CURRENCY = ['USD'] as const;
+export type ApplicationCurrency = (typeof APPLICATION_CURRENCY)[number];
+
+export const APPLICATION_CURRENCY_OPTIONS: Record<ApplicationCurrency, string> = {
+  USD: '$',
+} as const;
+
+export function useApplicationCurrency() {
+  const applicationCurrency = ref<ApplicationCurrency>('USD');
+
+  function setApplicationCurrency(value: ApplicationCurrency) {
+    // TODO: use local storage
+    applicationCurrency.value = value;
+  }
```

Comment:
[`useLocalStorage` | VueUse](https://vueuse.org/core/useLocalStorage/#uselocalstorage)

```suggestion
  const applicationCurrency = useLocalStorage<ApplicationCurrency>('app-currency', 'USD');

  function setApplicationCurrency(value: ApplicationCurrency) {
    applicationCurrency.value = value;
  }
```

Reply:
I also added useApplicationLanguage composable since they are similar.